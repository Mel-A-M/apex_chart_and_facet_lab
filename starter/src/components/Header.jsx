/**
 * Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved.
 * Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
 */

import React from 'react';
import PropTypes from 'prop-types';

/**
 * React component for rendering the header of the home page
 *
 * @param {string} companyTitle The company's title
 * @param {string} companyThumbnailUrl The URL of the company's icon
 * @param {string} aboutUrl The URL for the About link
 * @param {string} contactUrl The URL for the Contact Us link
 */
const Header = (props) => {
  const {
    companyTitle, companyThumbnailUrl, aboutUrl, contactUrl,
  } = props;

  return (
    <div className="logo" data-testid="Header">
      <img id="company-thumbnail" src={companyThumbnailUrl} alt="Company icon" />
      <h1 id="company-title">{companyTitle}</h1>
      <ul>
        <li><a id="about" href={aboutUrl}>About Us</a></li>
        <li><a id="contact" href={contactUrl}>Contact Us</a></li>
      </ul>
    </div>
  );
};

/*
 * Define the type of data used in this component.
 */
Header.propTypes = {
  companyTitle: PropTypes.string.isRequired,
  companyThumbnailUrl: PropTypes.string.isRequired,
  aboutUrl: PropTypes.string.isRequired,
  contactUrl: PropTypes.string.isRequired,
};

export default Header;
