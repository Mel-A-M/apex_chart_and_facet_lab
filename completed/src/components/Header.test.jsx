/* eslint-disable no-undef */
import React from 'react';

import { waitForElement, render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from './Header';

/** *******************************************************
 * These tests illustrate two different techniques to
 * validate the Header component
 */
describe('Test the rendering of the Header component', () => {
  /** *********************************************
   * Manually validate each element of the DOM
   ********************************************** */
  test('Example Validation of header DOM elements', async () => {
    /** *************
     * Act
     ************** */
    const {
      getByTestId,
      getByText,
      getByAltText,
    } = render(
      <Router path="/">
        <Header
          companyTitle="My Company"
          companyThumbnailUrl="http://www.somewhere.com/image.png"
          aboutUrl="http://www.somewhere.com/about"
          contactUrl="http://www.somewhere.com/contact"
        />
      </Router>,
    );


    /** *************
     * Assert
     ************** */
    await waitForElement(() => getByTestId('Header'));

    expect(getByText('My Company')).toBeTruthy();

    const img = getByAltText('Company icon');
    expect(img.src).toBe('http://www.somewhere.com/image.png');

    const about = getByText('About Us');
    expect(about).toBeTruthy();
    expect(about.href).toBe('http://www.somewhere.com/about');

    const contact = getByText('Contact Us');
    expect(contact).toBeTruthy();
    expect(contact.href).toBe('http://www.somewhere.com/contact');
  });

  /** *********************************************
   * Snapshot testing of the Header component
   ********************************************** */
  test('Validate Header snapshot', async () => {
  /** *************
   * Act
   ************** */
    const {
      container,
      getByTestId,
    } = render(
      <Router path="/">
        <Header
          companyTitle="My Company"
          companyThumbnailUrl="http://www.somewhere.com/image.png"
          aboutUrl="http://www.somewhere.com/about"
          contactUrl="http://www.somewhere.com/contact"
        />
      </Router>,
    );

    /** *************
     * Assert
     ************** */
    await waitForElement(() => getByTestId('Header'));
    expect(container).toMatchSnapshot();
  });
});
