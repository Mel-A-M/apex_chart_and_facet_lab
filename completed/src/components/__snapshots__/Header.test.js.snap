// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the rendering of the Header component Validate Header snapshot 1`] = `
<div>
  <div
    class="logo"
    data-testid="Header"
  >
    <img
      alt="Company icon"
      id="company-thumbnail"
      src="http://www.somewhere.com/image.png"
    />
    <h1
      id="company-title"
    >
      My Company
    </h1>
    <ul>
      <li>
        <a
          href="http://www.somewhere.com/about"
          id="about"
        >
          About Us
        </a>
      </li>
      <li>
        <a
          href="http://www.somewhere.com/contact"
          id="contact"
        >
          Contact Us
        </a>
      </li>
    </ul>
  </div>
</div>
`;
